Common/src/grid_movement/CSurfaceMovement.cpp

```diff
@@ -4796,5 +4940,93 @@ void CSurfaceMovement::WriteFFDInfo(CSurfaceMovement** surface_movement, CGeomet
     output_file.close();
 
   }
+}
+
+void CSurfaceMovement::SetDeformationMagnitude(CGeometry *geometry, CConfig *config, su2double factor, su2double** initialDef) {
+unsigned short iDV, iDV_Value;
+su2double dv_value;
+
+  for (iDV = 0; iDV < config->GetnDV(); iDV++) {
+    for (iDV_Value = 0; iDV_Value < config->GetnDV_Value(iDV); iDV_Value++) {
+      dv_value = initialDef[iDV][iDV_Value];
+      config->SetDV_Value(iDV, iDV_Value, dv_value*factor);
+    }
+  }
+}
+
+unsigned long CSurfaceMovement::calculateJacobianDeterminant(CGeometry *geometry, CConfig *config, CFreeFormDefBox *FFDBox){
+
+  su2double *ParamCoord, *CartCoord;
+  su2double determinant, d_du[3] = {0.0, 0.0, 0.0}, d_dv[3] = {0.0, 0.0, 0.0}, d_dw[3] = {0.0, 0.0, 0.0};
+  su2double CartCoords[3] = {0.0, 0.0, 0.0};
+  su2double Ba, Bb, Bc, Ba_der, Bb_der, Bc_der;
+
+  unsigned long iSurfacePoints;
+  unsigned short nDim = geometry->GetnDim();
+  unsigned short iMarker, iDim;
+  unsigned short iDegree, jDegree, kDegree, lDegree, mDegree, nDegree;
+  unsigned long negative_determinants = 0;
```

Comment:
It is better to declare variables locally as they become needed.

This way of declaring everything at the top is legacy from C and it is not a good C++ practice, other areas of the code have been extensively refactored to avoid this.